iotlabcli/helpers.py

```diff
@@ -188,15 +178,30 @@ def __setitem__(self, key, val):
             raise ValueError('Has different values for same key %r' % key)
 
     def add_file(self, file_path):
-        """Add a file to the dictionary. If None, do nothing """
+        """Add a file to the dictionary.
+        :param file_path the path of the file to add
+        :returns the id of the file in the dict
+        If a file with the same basename already exists inside,
+        then prefix with short hash is used
+        if None do nothing
+        """
         if file_path is None:
-            return
-        self[os.path.basename(file_path)] = read_file(file_path, 'b')
+            return None
+        key = os.path.basename(file_path)
+        value = read_file(file_path, 'b')
+        if key in self and self[key] != value:
+            # use md5 hash as prefix to handle duplicated basenames
+            # with different contents
+            key = md5(value) + '_' + key
+        self[key] = read_file(file_path, 'b')
+        return key
 
     def add_files_from_dict(self, keys, files_dict):
         """Add 'keys' files from 'files_dict' if present."""
         for key in keys:
-            self.add_file(files_dict.get(key, None))
+            file_element = files_dict.get(key, None)
+            if file_element is not None:
+                files_dict[key] = self.add_file(file_element)
```

Comment:
Why do you need to update `files_dict` there ?

From what I remember it is supposed to be only used as an input.
This would remove the special handling done here about checking if the file is empty, and remove the need for a special test to cover if `file_path` is None in `add_file`.

Reply:
Also, in there, you are replacing the content of `files_dict[key] == absolute_file_path` by `files_dict[key] == actual_key == (basename or md5_basename)` so not sure it is a wanted side effect.

Reply:
the file_path None as marker for "do nothing" was already there before I modified, I kept it out of caution, would have done differently

Reply:
Yes, I'm modifying files_dict. Which smelled bad to me, and I would love to do it differently. Might completely scrap the add_files_from_dict method actually

Reply:
It's been fixed, the modified dict is returned to the caller.